test/jdk/javax/swing/JFileChooser/6798062/bug6798062.java

```diff
@@ -21,54 +21,87 @@
-
-    private final JTextField tfLink = new JTextField();
-
-    private final JButton btnStart = new JButton("Start");
-
-    private final JButton btnStop = new JButton("Stop");
-
-    private final JButton btnGC = new JButton("Run System.gc()");
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JSlider;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+
+public class bug6798062 {
+
+    private static final String instructionsText = """
+            The test is suitable only for Windows.
+
+            1. Create a link
+            2. Copy path to the link into TextField
+            3. Run the Windows Task Manager. Select the Processes tab and find the java process
+            4. Press the Start button in the test window
+            5. Wait several minutes and observe in the Windows Task Manager
+               that Memory Usage of java process is not increasing
+            If memory usage is increasing, click Fail else click Pass .  """;
+
+    private static JSlider slider;
+    private static JFrame frame;
+    private static JTextField tfLink;
+    private static JButton btnStart;
+    private static JButton btnStop;
+    private static JButton btnGC;
 
     private ShellFolder folder;
-
     private Thread thread;
 
-    public static void main(String[] args) {
-        JFrame frame = new JFrame("bug6798062");
-
-        frame.setSize(400, 300);
-        frame.setLocationRelativeTo(null);
-        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
-        frame.add(new bug6798062().initialize());
-
-        frame.setVisible(true);
-    }
-
-    public void init() {
-        add(initialize());
+    public static void main(String[] args) throws Exception {
+         PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("JFileChooser Instructions")
+                .instructions(instructionsText)
+                .testTimeOut(5)
+                .rows(10)
+                .columns(35)
+                .build();
+
+        SwingUtilities.invokeAndWait(() -> {
+            slider = new JSlider(0, 100);
+            tfLink = new JTextField();
+            btnStart = new JButton("Start");
+            btnStop = new JButton("Stop");
+            btnGC = new JButton("Run System.gc()");
+            frame = new JFrame("bug6798062");
+
+            frame.setSize(400, 300);
+            frame.setLocationRelativeTo(null);
+            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+            frame.add(new bug6798062().initialize());
+
+            PassFailJFrame.addTestWindow(frame);
+            PassFailJFrame.positionTestWindow(frame,
+                    PassFailJFrame.Position.HORIZONTAL);
+            frame.setVisible(true);
+        });
+        passFailJFrame.awaitAndCheck();
     }
 
     private JComponent initialize() {
```

Comment:
I suggest creating the files in the current directory instead of `temp` or `home`. When run with jtreg, the current directory is set to `scratch` which is automatically removed by jtreg, which ensures no files are left behind if the test fails to clean them up for whatever reason.